### ISSUE TITLE
SyncType does not send changes when NetworkObject is despawned on the same tick it's spawned

### ISSUE THREAD

**Issue by ooonush** [FirstGearGames/FishNet#692]
**General**
Unity version: 6000.0.3
Fish-Networking version: 4.3.5

**Description**
SyncType does not send changes to client when NetworkObject is despawned on the same tick it's spawned.

**Replication**
1. Place the NetworkObject with the TestSpawner component on the scene:
```csharp
public class TestSpawner : NetworkBehaviour
{
    [SerializeField] private TestNetworkBehaviour _prefab;

    public override void OnStartClient()
    {
        if (IsHostInitialized)
        {
            NetworkObject no = NetworkManager.GetPooledInstantiated(_prefab.NetworkObject, true);
            Spawn(no);
            // and despawn this:
            Despawn(no);
        }
    }
}
```
2. Create a NetworkObject prefab with a TestNetworkBehaviour component:
```csharp
public class TestNetworkBehaviour : NetworkBehaviour
{
    private readonly SyncVar<int> _syncVar = new();

    public override void OnStartServer()
    {
        Debug.Log("OnStartServer");
        _syncVar.Value = 10;
    }

    public override void OnStartClient()
    {
        Debug.Log("OnStartClient " + _syncVar.Value);
    }
}
```
3. Start server
4. Start other clients
5. Start client-host

**Expected behavior**

On client started SyncType should be 10, not 0.
Since OnStartClient is being called, the SyncTypes should also be synchronized.
This is a problem for me when the projectile is despawned on the same tick it`s spawned.

**Screenshots**
![image](https://github.com/FirstGearGames/FishNet/assets/72870405/c27d3a7f-5abf-4fb1-8444-a7b62f9093bb)

**Comment by ooonush** [comment]
What works for me is this despawn method:

```csharp
private uint _tickSpawned;
private NetworkObject _no;

public override void OnStartClient()
{
    if (IsHostInitialized)
    {
        _no = NetworkManager.GetPooledInstantiated(_prefab.NetworkObject, true);
        Spawn(_no);
        _tickSpawned = TimeManager.Tick;
    }
}

private void TimeManager_OnTick()
{
    if (_no && TimeManager.Tick > _tickSpawned + 1)
    {
        Despawn(_no);
        _no = null;
    }
}
```

That is, I have to do a check for TimeManager.Tick > _tickSpawned + 1.

**Comment by ooonush** [comment]
For `clients-only` the check `TimeManager.Tick > _tickSpawned` works. But on the `client-host` I have to check `TimeManager.Tick > _tickSpawned + 2`, which is not very elegant.

**Comment by FirstGearGames** [comment]
I'm fairly certain your desired setup should work without any changes.
There's even code specific to sending pending SyncType data through on
despawn for this exact scenario.

I'll make this is a bug today and take a look, most likely something small.

**Comment by ooonush** [comment]
Also, I can't send an Rpc to a `client-host` with a NetworkObject for which Despawn was called right away.
Other clients do not have this error.

```csharp
[ObserversRpc]
private void EndActionObserversRpc(NetworkActionBase action)
{
    action.EndActionClient(); // Null Reference
}
```

**Comment by FirstGearGames** [comment]
Do not have time to get this into 4.3.6 but making it high priority for 4.3.7

**Comment by FirstGearGames** [comment]
Priority updated to after delta serializers for prediction. The serializers are close to being finished so I'd like to finish them up first.

**Comment by FirstGearGames** [comment]
Resolved in 4.4.0
